standard/variables.md

```diff
@@ -901,3 +901,139 @@ variable_reference
 ## 9.6 Atomicity of variable references
 
 Reads and writes of the following data types shall be atomic: `bool`, `char`, `byte`, `sbyte`, `short`, `ushort`, `uint`, `int`, `float`, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list shall also be atomic. Reads and writes of other types, including `long`, `ulong`, `double`, and `decimal`, as well as user-defined types, need not be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.
+
+## §ref-span-safety Reference variables and returns
+
+### §ref-span-safety-general General
+
+A *reference variable* is a local variable or `struct` declared with the `ref` modifier. A ref local does not create a new storage location. Instead, a ref ref local represents the same storage location as its initializing expression. Thus, the value of a reference variable is always the same as the underlying variable.
+
+A `ref struct` is a `struct` whose fields include a `ref` struct or a ref-like field. A ref-like field does not create a new storage location. Instead, if refers to the same storage as its initializing expression. Ref struct types include `Span<T>`, `ReadOnlySpan<T>`, and other types that include an unmanaged pointer as a member.
+
+A *reference return* is the expression returned by reference from a method whose return type includes the `ref` or `ref readonly` modifiers (§14.6.1). The return refers to the same storage as its expression.
+
+### §ref-span-safety-escape-scopes Safe to escape scopes
+
+At compile-time, each expression is associated with a scope that expression is permitted to escape to, *safe-to-escape-scope*. Each variable is associated with the scope a reference to it is permitted to escape to, *ref-safe-to-escape-scope*. For a given variable expression, these may be different. When the *ref-safe-to-escape* scope is *calling-method*, the variable is *safe-to-return*. A variable that is *safe-to-return* may escape the enclosing method as a whole. In other words, the variable can *returned-by-ref*.
```

Comment:
The first sentence refers to "each expression", the second refers to "each variable", and the third refers to "a given variable expression". Are the sentences referring to the same items? If so, consider using a common name - perhaps "expression", or if this is specifically about `ref` cases, perhaps "ref expression". (Are the expressions always variables or is the expression in `return ref F();` also an example?)

> When the *ref-safe-to-escape* scope is *calling-method*, the variable is *safe-to-return*. A variable that is *safe-to-return* may escape the enclosing method as a whole. In other words, the variable can *returned-by-ref*.

When the ref-safe-to-escape scope is _calling-method_, the variable is safe-to-return _by-ref_. (When the safe-to-escape scope is _calling-method_, the variable is safe-to-return _by-value_.)

Reply:
I need to think about this one more. This was intentional, but it's awkward. Not all expressions can be ref assigned or returned. Only variables (which have defined storage) can be assigned to a ref vaiable.

I'll keep thinking on the wording here.